### ISSUE TITLE
How to dump batchnorm jit kernel

### ISSUE THREAD

**Issue by zhewang1-intc** [oneapi-src/oneDNN#1425]
Hi intel oneDNN team.
I want to investigate the batchnorm calculate method based on AVX512 ISA , so i need to dump the batchnorm jit kernel.
After i compiled the source code , i entered this dir `oneDNN/build/examples` and execute the command `ONEDNN_JIT_DUMP=1 ./primitives-batch-normalization-cpp`
However , there are no bin files be dumped. But if i test some other examples such as eltwise `ONEDNN_JIT_DUMP=1 ./primitives-eltwise-cpp` , it can generate some bin files like `dnnl_dump_cpu_jit_uni_kernel.0.bin` successfully.
So how can i dump the batchnorm jit kernel?

**Comment by densamoilov** [comment]
Can you run batch normalization with DNNL_VERBOSE=1 and share the output? It might happen that a reference implementation is executed.

**Comment by zhewang1-intc** [comment]
> 
the outputs are as follow:
![image](https://user-images.githubusercontent.com/72838274/179431973-14660610-6665-4da7-b126-6b1f1864192b.png)
by the way, i tried to dump the jit kernel via batchnorm gtest, it works. i found only the memory layout is some formats(e.g nhwc), the batchnorm will call the jit version. But if oneDNN process some other memory layout(e.g nchw), it will not use AVX512 feature.
so i wonder dose the performance of nchw batchnorm is good enough?
why we do not add a nchw version of batchnorm jit kernel?

**Comment by vpirogov** [comment]
@zhewang1-intc, the implementation dispatched in this case is pre-compiled, not JITed. You can find the source code here: [simple_layer_normalization.hpp](https://github.com/oneapi-src/oneDNN/blob/master/src/cpu/simple_layer_normalization.cpp)

**Comment by zhewang1-intc** [comment]
> @zhewang1-intc, the implementation dispatched in this case is pre-compiled, not JITed. You can find the source code here: [simple_layer_normalization.hpp](https://github.com/oneapi-src/oneDNN/blob/master/src/cpu/simple_layer_normalization.cpp)

I saw the implementation, but i think the performance of pre-compiled solution is not as good as JIT solution, so i wonder why oneDNN didn't implement a JIT kernel for this case?

**Comment by densamoilov** [comment]
Layer normalization implementation is a bit different from others because it combines two implementations: 1) pre-compiled 2) JIT. Depending on target hardware and the given problem the layer normalization implementation decides which implementation to select. 

In general, decision on adding new optimizations in oneDNN is based on whether or not it demonstrates value, e.g. accelerates a particular workload. Therefore JIT implementations usually have more narrow coverage then their reference counterparts.

**Comment by zhewang1-intc** [comment]
> 

ok, got it. I can close this issue now, thanks for your help intel oneDNN team!
